HBASE-23937 : Support Online LargeLogs similar to SlowLogs APIs

virajjasani opened the pull request.

virajjasani commented:
@ndimiduk Could you please review this patch? This is one of the sub-tasks that you suggested i.e. support large logs API similar to slow logs.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 17s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m 51s |  master passed  |
| +1 :green_heart: |  spotbugs  |   8m 21s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 52s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 44s |  the patch passed  |
| -0 :warning: |  rubocop  |   0m 13s |  The patch generated 4 new + 339 unchanged - 0 fixed = 343 total (was 339)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  12m 31s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   3m  5s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   8m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 56s |  The patch does not generate ASF License warnings.  |
|  |   |  62m 43s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle cc hbaseprotoc prototool rubocop |
| uname | Linux 368a8adcb0a2 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / eed730e492 |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 83 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) spotbugs=3.1.12 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 41s |  master passed  |
| +1 :green_heart: |  compile  |   3m 44s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m  1s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 26s |  hbase-client in master failed.  |
| -0 :warning: |  javadoc  |   0m 40s |  hbase-server in master failed.  |
| -0 :warning: |  javadoc  |   0m 49s |  hbase-thrift in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   6m 17s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 40s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 40s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 52s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 27s |  hbase-client in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 40s |  hbase-server in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 51s |  hbase-thrift in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 58s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 59s |  hbase-client in the patch passed.  |
| -0 :warning: |  unit  |  68m 43s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  unit  |   2m 32s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  unit  |   7m 10s |  hbase-shell in the patch passed.  |
|  |   | 121m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 613a68f136e9 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / eed730e492 |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-thrift.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/testReport/ |
| Max. process+thread count | 6155 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

virajjasani commented on hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java:
```diff
@@ -2276,6 +2276,18 @@ boolean snapshotCleanupSwitch(final boolean on, final boolean synchronous)
+   * @throws IOException if a remote or network exception occurs
+   */
+  List<SlowLogRecord> getLargeLogResponses(final Set<ServerName> serverNames,
+    final SlowLogQueryFilter largeLogQueryFilter) throws IOException;
```
@bharathv I am planning to change `SlowLogRecord` to `OnlineLogRecord` and `SlowLogQueryFilter` to `OnlineLogQueryFilter`. Will make that change before merging the PR because this class is being referred at multiple places (in thrift), so will be redundant refactor for now (to make review relevant to actual changes).

Also, the overall change is providing new API in Admin and shell command and to have 2 flags at server side (isSlowLog and isLargeLog) to fetch relevant records from server.

bharathv commented on hbase-protocol-shaded/src/main/protobuf/Admin.proto:
```diff
@@ -287,6 +287,8 @@ message SlowLogResponseRequest {
   optional string client_address = 3;
   optional string user_name = 4;
   optional uint32 limit = 5 [default = 10];
+  optional bool is_slow_log = 6 [default = false];
```
use an enum? OnlineLogType for ex.. Also, you don't need to have two separate RPCs if you pass this "type" as a param to the RPC call (we can potentially have other "types" in the future in which case we don't want to add a new set of RPCs).

bharathv replied:
Another thought..why not implement this type as a LogQueryFilter? that makes it much simpler right?

virajjasani replied:
IMHO having separate shell command and Admin API should be good and clean for operators rather than having "type" as a filter. I agree on implementing enum for "type" on server side including this proto file. We can have filter logic at server side but having separate API for client would be better because all filters that we want to provide to client should be related to actual data filters - client, user, region, table etc rather than type.

Most of the time, user would not use filter so we don't want them to use filter just for log type.
`get_slowlog_responses '*'` and `get_largelog_responses '*'` might be mostly used (without filters).

bharathv replied:
> IMHO having separate shell command and Admin API should be good and clean for operators rather than having "type" as a filter. 

Umm.. shell command is fine, probably makes it easier for operators...whats the use of having a separate Admin API? All of these requests fall under the same "get_log" category, so probably good to logically group them into same API (which takes Type parameter) and avoid code bloat?

virajjasani replied:
I believe we follow 1:1 shell command to Admin API right? This is to ensure operator using shell commands and Java/Thrift clients using Admin API have similar experience. No?

bharathv replied:
They still have the same experience, its just what each client does with the API. For example, there are 10 (or more) ways of creating a table from shell but only 3 corresponding create admin APIs. All these 10 map to one of these 3 APIs. Java/Thrift clients for this API build the TableDescriptor and use one of these 3 API calls. It is somewhat a similar case here. At least this is my understanding.

virajjasani replied:
Hmm yeah makes sense. I am not too strong of an opinion to keep 1:1 for slow and large logs. Just thought of keeping them same but yes having single API with LogQueryFilter to differentiate between slow and large logs seems better approach.

bharathv commented on hbase-shell/src/main/ruby/shell/commands/get_largelog_responses.rb:
```diff
@@ -0,0 +1,78 @@
+e.g.: host187.example.com,60020,1289493121758 (find servername in
+master ui or when you do detailed status in shell)
+
+Provide optional filter parameters as Hash.
```
This can probably be simplified too with 'TYPE' => LARGE_LOGS

virajjasani replied:
Same as above. Let's have type filter for server side logic but have clean, separate shell command and Admin API. Anyways, code would not be too much and would be simple for client.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 40s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 34s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m  3s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m 56s |  master passed  |
| +1 :green_heart: |  spotbugs  |   8m 37s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 35s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 51s |  the patch passed  |
| -0 :warning: |  rubocop  |   0m 15s |  The patch generated 4 new + 339 unchanged - 0 fixed = 343 total (was 339)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  12m 53s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   4m  5s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |  13m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   1m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  71m 18s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle cc hbaseprotoc prototool rubocop |
| uname | Linux 219c476a6f4c 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fded2b9ddc |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 83 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) spotbugs=3.1.12 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 38s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m  5s |  master passed  |
| +1 :green_heart: |  compile  |   3m 22s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 27s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m  3s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 46s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 15s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 17s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 57s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 47s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  0s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 107m  4s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 43s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  unit  |   7m  9s |  hbase-shell in the patch passed.  |
|  |   | 156m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7e5b4f32760c 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fded2b9ddc |
| Default Java | 1.8.0_232 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/testReport/ |
| Max. process+thread count | 4312 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 38s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 52s |  master passed  |
| +1 :green_heart: |  compile  |   3m 46s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 48s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 25s |  hbase-client in master failed.  |
| -0 :warning: |  javadoc  |   0m 40s |  hbase-server in master failed.  |
| -0 :warning: |  javadoc  |   0m 48s |  hbase-thrift in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   6m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 50s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 50s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 32s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 41s |  hbase-client in the patch failed.  |
| -0 :warning: |  javadoc  |   1m  7s |  hbase-server in the patch failed.  |
| -0 :warning: |  javadoc  |   1m 20s |  hbase-thrift in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 26s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 34s |  hbase-client in the patch passed.  |
| -0 :warning: |  unit  | 104m 35s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  unit  |   2m 38s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  unit  |   7m 11s |  hbase-shell in the patch passed.  |
|  |   | 163m 52s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 451d10f1a300 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / fded2b9ddc |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-thrift.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/testReport/ |
| Max. process+thread count | 5506 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

bharathv commented on hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java:
```diff
@@ -3952,17 +3952,27 @@ private void getProcedureResult(long procId, CompletableFuture<Void> future, int
-      .map(CompletableFuture::join)
-      .flatMap(List::stream)
-      .collect(Collectors.toList()));
+    if (slowLogQueryFilter.getType() == null
```
Why this branch on the client? Can't the server side infer the type and respond accordingly?

virajjasani replied:
Nope, both can't use same enum because at server side, we have enum in TooSlowLog.proto which we save in Queue, whereas this enum is for client side filter object.

Having this branch on client side is better compared to passing it on to server because otherwise we will need to have it in Admin.proto too to pass it over in RPC call (too many places).

bharathv commented on hbase-protocol-shaded/src/main/protobuf/TooSlowLog.proto:
```diff
@@ -42,4 +42,12 @@ message SlowLogPayload {
+
+  enum Type {
+    SLOW_LOG = 0;
+    LARGE_LOG = 1;
```
nit: Helps to have a code comment about what a SLOW log is and what a LARGE log is..

virajjasani replied:
Done

bharathv commented on hbase-protocol-shaded/src/main/protobuf/Admin.proto:
```diff
@@ -369,6 +369,9 @@ service AdminService {
   rpc GetSlowLogResponses(SlowLogResponseRequest)
     returns(SlowLogResponses);
 
+  rpc GetLargeLogResponses(SlowLogResponseRequest)
```
same question as above.. we can get rid of this one, right? (I thought that was our discussion in the PR earlier).

virajjasani replied:
We discussed about getting rid of separate Admin API and have a same endpoint. This one is for RS RPC services. Since we don't want to have multiple enums in Admin.proto, TooSlowLogs.proto, we need this separately and it's only used in RSRpcServices with minimal code.

virajjasani replied:
Another reason why we must have this is to find out sometimes which specific Get API was slow or large in nature even between get_slow and get_large responses.

Sample response:
```
{
  "startTime": 1585735414657,
  "processingTime": 1,
  "queueTime": 0,
  "responseSize": 2351,
  "clientAddress": "171.30.20.8:63571",
  "serverClass": "HRegionServer",
  "methodName": "GetLargeLogResponses",
  "callDetails": "GetLargeLogResponses(org.apache.hadoop.hbase.shaded.protobuf.generated.AdminProtos$SlowLogResponseRequest)",
  "param": "class org.apache.hadoop.hbase.shaded.protobuf.generated.AdminProtos$SlowLogResponseRequest",
  "userName": "vjasani"
}
{
  "startTime": 1585735372313,
  "processingTime": 1,
  "queueTime": 0,
  "responseSize": 2260,
  "clientAddress": "171.30.20.8:63571",
  "serverClass": "HRegionServer",
  "methodName": "GetSlowLogResponses",
  "callDetails": "GetSlowLogResponses(org.apache.hadoop.hbase.shaded.protobuf.generated.AdminProtos$SlowLogResponseRequest)",
  "param": "class org.apache.hadoop.hbase.shaded.protobuf.generated.AdminProtos$SlowLogResponseRequest",
  "userName": "vjasani"
}
```

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  1s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 28s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m 47s |  master passed  |
| +1 :green_heart: |  spotbugs  |   8m  2s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 28s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 46s |  the patch passed  |
| -0 :warning: |  rubocop  |   0m 13s |  The patch generated 6 new + 338 unchanged - 1 fixed = 344 total (was 339)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 30s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   2m 57s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   8m 38s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 52s |  The patch does not generate ASF License warnings.  |
|  |   |  58m 16s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle cc hbaseprotoc prototool rubocop |
| uname | Linux fdf83ae67f71 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e5a8435b48 |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 83 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) spotbugs=3.1.12 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 34s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 18s |  master passed  |
| +1 :green_heart: |  compile  |   3m  4s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 46s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 56s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  7s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  7s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 47s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 57s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 45s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 51s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |  65m 36s |  hbase-server in the patch passed.  |
| -1 :x: |  unit  |   2m 30s |  hbase-thrift in the patch failed.  |
| +1 :green_heart: |  unit  |   7m 11s |  hbase-shell in the patch passed.  |
|  |   | 111m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 6099aecb0b0e 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e5a8435b48 |
| Default Java | 1.8.0_232 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-hbase-thrift.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/testReport/ |
| Max. process+thread count | 5690 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m 42s |  master passed  |
| +1 :green_heart: |  compile  |   4m  6s |  master passed  |
| +1 :green_heart: |  shadedjars  |   7m 24s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 31s |  hbase-client in master failed.  |
| -0 :warning: |  javadoc  |   0m 47s |  hbase-server in master failed.  |
| -0 :warning: |  javadoc  |   1m  6s |  hbase-thrift in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 19s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 19s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 18s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 29s |  hbase-client in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 49s |  hbase-server in the patch failed.  |
| -0 :warning: |  javadoc  |   1m  1s |  hbase-thrift in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 12s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 10s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |  85m  4s |  hbase-server in the patch passed.  |
| -0 :warning: |  unit  |   3m  4s |  hbase-thrift in the patch failed.  |
| +1 :green_heart: |  unit  |   7m 15s |  hbase-shell in the patch passed.  |
|  |   | 145m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux baf1a5acc1c4 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e5a8435b48 |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-thrift.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-thrift.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/testReport/ |
| Max. process+thread count | 5685 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/4/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

virajjasani commented:
@bharathv addressed all comments and renamed classes as I mentioned in initial comment. Waiting for QA before merging. If there are any specific concern, I can take it up in follow up Jira.
Thanks

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  1s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 21s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m 53s |  master passed  |
| +1 :green_heart: |  spotbugs  |   7m 23s |  master passed  |
| -0 :warning: |  patch  |   2m  1s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  3s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 35s |  hbase-client: The patch generated 1 new + 290 unchanged - 0 fixed = 291 total (was 290)  |
| -0 :warning: |  checkstyle  |   1m 11s |  hbase-server: The patch generated 1 new + 77 unchanged - 0 fixed = 78 total (was 77)  |
| -0 :warning: |  checkstyle  |   0m 45s |  hbase-thrift: The patch generated 1 new + 1 unchanged - 0 fixed = 2 total (was 1)  |
| -0 :warning: |  rubocop  |   0m 11s |  The patch generated 5 new + 338 unchanged - 1 fixed = 343 total (was 339)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  10m 34s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2.  |
| +1 :green_heart: |  hbaseprotoc  |   2m 57s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   8m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 59s |  The patch does not generate ASF License warnings.  |
|  |   |  56m  9s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle cc hbaseprotoc prototool rubocop |
| uname | Linux 36044977adbe 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 52c101389e |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-general-check/output/diff-checkstyle-hbase-client.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-general-check/output/diff-checkstyle-hbase-thrift.txt |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 93 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) spotbugs=3.1.12 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 16s |  master passed  |
| +1 :green_heart: |  compile  |   3m  6s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 46s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 57s |  master passed  |
| -0 :warning: |  patch  |   7m 36s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   6m  9s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 57s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 55s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 40s |  hbase-thrift generated 1 new + 12 unchanged - 1 fixed = 13 total (was 13)  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 58s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  0s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |  67m 10s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 22s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  unit  |   7m  5s |  hbase-shell in the patch passed.  |
|  |   | 116m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7126012a8d2a 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 52c101389e |
| Default Java | 1.8.0_232 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk8-hadoop2-check/output/diff-javadoc-javadoc-hbase-thrift.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/testReport/ |
| Max. process+thread count | 5596 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 53s |  master passed  |
| +1 :green_heart: |  compile  |   3m 37s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 39s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 25s |  hbase-client in master failed.  |
| -0 :warning: |  javadoc  |   0m 39s |  hbase-server in master failed.  |
| -0 :warning: |  javadoc  |   0m 49s |  hbase-thrift in master failed.  |
| -0 :warning: |  patch  |   8m 54s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 38s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 38s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 33s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 25s |  hbase-client in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 39s |  hbase-server in the patch failed.  |
| -0 :warning: |  javadoc  |   0m 49s |  hbase-thrift in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 58s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 58s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |  73m 42s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 29s |  hbase-thrift in the patch passed.  |
| +1 :green_heart: |  unit  |   7m  9s |  hbase-shell in the patch passed.  |
|  |   | 124m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.8 Server=19.03.8 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1346 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 6e4b73c75d45 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 52c101389e |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-thrift.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-client.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-thrift.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/testReport/ |
| Max. process+thread count | 6230 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-thrift hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1346/6/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.
